Remotion/Data/DomainObjects.UnitTests/Persistence/Rdbms/SqlServer/DbCommandBuilders/SqlDbCommandBuilderFactoryTest.cs

```diff
@@ -159,12 +160,21 @@ public void CreateForSelect_UnionView ()
     }
 
     [Test]
-    public void CreateForQuery ()
+    public void CreateForQuery_TextStatementType_ReturnsSqlQueryDbCommandBuilder ()
     {
-      var result = _factory.CreateForQuery("statement", new QueryParameterWithDataParameterDefinition[0]);
+      var result = _factory.CreateForQuery(QueryStatementType.Text, "statement", new QueryParameterWithDataParameterDefinition[0]);
 
-      Assert.That(result, Is.TypeOf(typeof(SqlQueryDbCommandBuilder)));
-      Assert.That(((SqlQueryDbCommandBuilder)result).SqlDialect, Is.SameAs(_sqlDialectStub.Object));
+      Assert.That(result, Is.TypeOf(typeof(QueryDbCommandBuilder)));
+      Assert.That(((QueryDbCommandBuilder)result).SqlDialect, Is.SameAs(_sqlDialectStub.Object));
+    }
+
+    [Test]
+    public void CreateForQuery_StoredProcedureStatementType_ReturnsStoredProcedureDbCommandBuilder ()
+    {
+      var result = _factory.CreateForQuery(QueryStatementType.StoredProcedure, "statement", new QueryParameterWithDataParameterDefinition[0]);
+
+      Assert.That(result, Is.TypeOf(typeof(StoredProcedureDbCommandBuilder)));
```

Comment:
note: you can use the generic syntax, too: `Is.TypeOf<...>` 
personal preferences apply, I like the generic because it's shorter and easier to type/less wordy.